pyiceberg/manifest.py

```diff
@@ -620,6 +623,13 @@ def fetch_manifest_entry(self, io: FileIO, discard_deleted: bool = True) -> List
             ]
 
 
+@cached(cache=LRUCache(maxsize=128), key=lambda io, manifest_list: hashkey(manifest_list))
```

Comment:
Did the use of `cachetools` LRUCache, instead of `functools.lru_cache` solve the issues with the FileSystem on M1 Macs we were observing before?

Reply:
yep! the difference is that `functools.lru_cache` uses all of the function's args as the cache key, including the `io` arg. 
Using `cachetools`, I can specify which argument to use as the cache key. In this case, only using `manifest_list` as the cache key
```
key=lambda io, manifest_list: hashkey(manifest_list)
```

Reply:
That's so clever - so the issue was with the function arguments being 'held up' by the cache preventing them from being GCed 🤯 🧠 

This is next level @kevinjqliu 💯

Reply:
That's my best theory, haha. Regardless of the underlying cause, we should not be using `io` as the cache key!